### PR TITLE
feat: add `sanity_check implementation` for block headers

### LINKED ISSUE #5800: feat: add `sanity_check` implementation for block headers
### Describe the feature

Sometimes, we need to protect the system against potential DOS attack. In this sense, it could be nice to have a method to ensure the validity of critical values within the block header. We could implement a global sanity check method for the `Header` structure, similar to the approach demonstrated [here](https://github.com/ethereum/go-ethereum/blob/f1794ba2788baf34489847bfa9ca00e067507db0/core/types/block.go#L134-L152).

While our block numbers (natively u64) already enforce one aspect, expanding this with additional tests for other header values accompanied by dedicated error handling could significantly enhance our system's security.

### Proposed Changes:
- Introduce a global sanity check method for the `Header` structure.
- Add tests for various header values with dedicated error handling.

These changes are aimed at strengthening our system's defenses and ensuring the integrity of block header data.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #5800.

### CODE REVIEW

File: crates/primitives/src/header.rs [hunk 1]
```diff
@@ -117,6 +126,44 @@ impl Default for Header {
...
+        }
+        Ok(())
+    }
+
+    /// Performs a sanity check on the block difficulty field of the header.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the block difficulty exceeds 80 bits.
+    pub fn check_difficulty(&self) -> Result<(), HeaderError> {
```
Comment (mattsse) [inline crates/primitives/src/header.rs:146]: naming nit:

these should be called `ensure_`

File: crates/primitives/src/header.rs [hunk 2]
```diff
@@ -117,6 +126,44 @@ impl Default for Header {
...
+
+    /// Performs combined sanity checks on multiple header fields.
+    ///
+    /// This method combines checks for block difficulty and extradata sizes.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if either the block difficulty exceeds 80 bits
+    /// or if the extradata size is larger than 100 KB.
+    pub fn sanity_check(&self) -> Result<(), HeaderError> {
```
Comment (mattsse) [inline crates/primitives/src/header.rs:161]: this function name is a bit ambiguous. 

perhaps `ensure_well_formed`? or something?

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, supportive, only naming nits

File: crates/primitives/src/header.rs (hunk 1 above)
Comment (tcoratger) [reply crates/primitives/src/header.rs]: done

File: crates/primitives/src/header.rs (hunk 2 above)
Comment (tcoratger) [reply crates/primitives/src/header.rs]: done

**Review by mattsse — APPROVED** [review]

**Pull request merged**